projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -41,87 +42,97 @@ export class IconComponent {
    * icon type changes at run time.
    */
-  protected styleClasses: string[];
+  @HostBinding('class') styleClasses: string[] = [];
+
+  private iconValue?: string;
+  private iconResourceType: IconResourceType;
 
   constructor(
     protected iconLoader: IconLoaderService,
-    protected elementRef: ElementRef<HTMLElement>,
-    protected renderer: Renderer2
+    protected sanitizer: DomSanitizer
   ) {}
 
-  protected setIcon(type: ICON_TYPE): void {
+  protected setIcon(type: ICON_TYPE_STRING): void {
     if (!type || <string>type === '') {
       return;
     }
-    this.icon = this.iconLoader.getHtml(type);
-    this.addStyleClasses(type);
+    this.iconResourceType = this.iconLoader.getResourceType(type);
+    if (this.isSvgIcon()) {
+      // looks like angular does not sanitize URLs when binding to the xlink:href attribute of svg elements
+      // so we need to sanitze the URL manually
+      this.iconValue =
+        this.sanitizer.sanitize(
+          SecurityContext.URL,
+          this.iconLoader.getSvgPath(type) || null
+        ) || undefined;
+    } else if (this.isTextIcon()) {
+      this.iconValue = this.iconLoader.getSymbol(type);
+    } else {
+      // assume IconResourceType.LINK by default
+      this.iconValue = undefined;
+    }
     this.iconLoader.addLinkResource(type);
-    this.flipIcon(type);
+    this.addStyleClasses(type);
+  }
+
+  public get icon(): string | undefined {
+    return this.iconValue;
+  }
+
+  public isSvgIcon(): boolean {
+    return this.iconResourceType === IconResourceType.SVG;
+  }
+
+  public isTextIcon(): boolean {
+    return this.iconResourceType === IconResourceType.TEXT;
   }
 
   /**
-   * The icons supports flipping for some icons to support rtl and ltr directions.
+   * If necessary, adds CSS classes for flipping for some icons to support rtl and ltr directions.
    */
-  protected flipIcon(type: ICON_TYPE) {
-    // TODO: this can be dropped with the next major release.
+  protected flipIcon(type: ICON_TYPE_STRING) {
+    // TODO: this check can be dropped with the next major release.
     if (!this.iconLoader.getFlipDirection) {
       return;
     }
     const iconDirection = this.iconLoader.getFlipDirection(type);
-    this.flipAtLtr = iconDirection === DirectionMode.LTR;
-    this.flipAtRtl = iconDirection === DirectionMode.RTL;
+    if (iconDirection === DirectionMode.LTR) {
+      this.styleClasses.push(CSS_CLASSES.LTR);
+    } else if (iconDirection === DirectionMode.RTL) {
+      this.styleClasses.push(CSS_CLASSES.RTL);
+    }
   }
 
   /**
-   * Adds the style classes and the link resource (if available).
+   * Adds the style classes for the icon.
    */
-  protected addStyleClasses(type: ICON_TYPE): void {
-    this.renderer.addClass(this.host, 'cx-icon');
-
-    this.styleClasses?.forEach((cls) =>
-      this.renderer.removeClass(this.host, cls)
-    );
+  protected addStyleClasses(type: ICON_TYPE_STRING): void {
+    this.styleClasses = [CSS_CLASSES.CX_ICON];
```

Comment:
For some reason it works best to assign a brand new array to `styleClasses`. Safes us the trouble of deleting individual classes.

For some reason, Angular still retains the classes that are declared in the html template of the host component. I have not idea why. But tests indicate it, see `'host icon components'` specs.